### DISCUSSION TITLE
No nullable in SQL but nullable in Dataclass ?

### DISCUSSION THREAD

**Discussion by CharlotteJu** [simolus3/drift#2556] [Q&A] [answered]
Hi,

I'm using Drift in a form application, and I'm having an issue with columns that are non-nullable in the SQL script but I need the generated class to accept a null type.

Here's an example to be clearer:

I have a table:

```
class User extends Table {
  IntColumn get function => integer().autoIncrement()();
  IntColumn get parentId => integer().nullable();
  TextColumn get lastName => integer()();
  TextColumn get firstName => integer()();
}
```
My database expects lastName and firstName not to be null, but when the user fills out the form, I need to be able to store the value of lastName in an object before they provide their firstName.

Is there a way to create a second constructor in the generated DataClass for this purpose? 
Or should I rather write:
`TextColumn get lastName => integer().check(lastName.isNotNull()).nullable()();`
?
Or should I create another object myself and then cast it?

Thank you in advance, and thank you for this excellent library!

**Comment by simolus3** [comment, marked as answer]
>  Or should I create another object myself and then cast it?

Yup, you'll want to use a [companion](https://drift.simonbinder.eu/docs/getting-started/writing_queries/#updates-and-deletes) class that drift generates for you.

The reason `function` is non-nullable is because it would be annoying for selects otherwise. Then you'd need `!` everywhere after selecting from the database, where you know that the column must be non-nullable. The companion class allows you to directly specify which fields have been set.

**Reply by CharlotteJu** [reply]
Thank's Simon !!
